src/common/notifications.cpp

```diff
@@ -0,0 +1,89 @@
+#include "pch.h"
+#include "notifications.h"
+
+#include <unknwn.h>
+#include <winrt/base.h>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Data.Xml.Dom.h>
+#include <winrt/Windows.Foundation.Collections.h>
+#include <winrt/Windows.UI.Notifications.h>
+#include <winrt/Windows.ApplicationModel.Background.h>
+
+using namespace winrt::Windows::ApplicationModel::Background;
+using winrt::Windows::Data::Xml::Dom::XmlDocument;
+using winrt::Windows::UI::Notifications::ToastNotification;
+using winrt::Windows::UI::Notifications::ToastNotificationManager;
+
+namespace
+{
+    constexpr std::wstring_view TASK_NAME = L"PowerToysBackgroundNotificationsHandler";
+    constexpr std::wstring_view TASK_ENTRYPOINT = L"PowerToysNotifications.BackgroundHandler";
+    constexpr std::wstring_view APPLICATION_ID = L"PowerToys";
+}
+
+void notifications::register_background_toast_handler()
+{
+    try
+    {
+        // Re-request access to clean up from our previous versions
+        BackgroundExecutionManager::RemoveAccess();
+        BackgroundExecutionManager::RequestAccessAsync().get();
+
+        BackgroundTaskBuilder builder;
+        ToastNotificationActionTrigger trigger{ APPLICATION_ID };
+        builder.SetTrigger(trigger);
+        builder.TaskEntryPoint(TASK_ENTRYPOINT);
+        builder.Name(TASK_NAME);
+
+        const auto tasks = BackgroundTaskRegistration::AllTasks();
+        const bool already_registered = std::any_of(begin(tasks), end(tasks), [=](const auto& task) {
+            return task.Value().Name() == TASK_NAME;
+        });
+        if (already_registered)
+        {
+            return;
+        }
+        (void)builder.Register();
+    }
+    catch (...)
+    {
+        // Couldn't register the background task, nothing we can do
+    }
+}
+
+void notifications::show_toast(std::wstring_view message)
+{
+    // The toast won't be actually activated in the background, since it doesn't have any buttons
```

Comment:
The comment is kind of cryptic. It seems it's only needed because the function name is implying something that it shouldn't imply, or maybe I completely missed the point.

Reply:
This comment basically means that we can implement `show_toast` in terms of `show_toast_background_activated` safely.